edx_sysadmin/api/views.py

```diff
@@ -0,0 +1,72 @@
+import json
+import logging
+
+from django.conf import settings
+from django.utils.decorators import method_decorator
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from edx_sysadmin.api.decorators import authenticate_github_request
+from edx_sysadmin.git_import import add_repo
+from edx_sysadmin.utils.utils import (
+    get_local_active_branch,
+    get_local_course_repo,
+)
+
+from git import Repo, InvalidGitRepositoryError, NoSuchPathError
+
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(authenticate_github_request(), name="post")
+class GitReloadAPIView(APIView):
+    """
+    APIView to reload courses from github on triggering github webhook
+    """
+
+    def post(self, request):
+        """
+        Trigger for github webhooks for course reload
+        """
+        message = ""
+        try:
+            if request.headers.get("X-Github-Event") == "push":
+                payload = json.loads(request.POST.get("payload"))
+                repo_name = payload["repository"]["name"]
+
+                repo = get_local_course_repo(repo_name)
+                if repo:
```

Comment:
nit: Instead of these nested `if/else` I'd prefer placing these check at the top of view that makes them more readable since we have complete check code in one place.
```
if not repo_ssh_url:
    logger.exception("GitReloadAPIView:: Request Payload is not appropriate. ssh_url is missing")
```
and exception/log messages should be prefixed with view and should explain the reason of exception.

Reply:
Updated.